### PR TITLE
Remove use of CustomData and make use of new Fragment for AdapterId

### LINKED ISSUE #293: Remove use of CustomData in the code
The files listed below have been detected as having code using `CustomData`. This detection has been automatic, and may have picked up on code comments or documentation. Please review each file carefully and remove all instances of `CustomData`.

This should be completed by the 30th October 2020.

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Elements\ToBar.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Elements\ToEdge.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Elements\ToNode.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Elements\ToPanel.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToAreaTemperatureLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToAreaUniformlyDistributedLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToBarDistributedLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToBarPointLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToBarTemperatureLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToBarUniformlyDistributedLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToGravityLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToLoadcase.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToLoadCombination.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToPointDisplacement.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Loads\ToPointLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Properties\ToConstraint4DOF.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Properties\ToConstraint6DOF.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Properties\ToMaterial.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Properties\ToMeshSettings1D.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Properties\ToMeshSettings2D.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Properties\ToSectionProperty.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Convert\ToBHoM\Properties\ToSurfaceProperty.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Create.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Elements\Edge.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Elements\Line.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Elements\Point.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Elements\Surface.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Loads\BarDistributedLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Loads\BarPointLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Loads\ConcentratedLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Loads\DistributedLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Loads\GravityLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Loads\Loadcase.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Loads\LoadCombination.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Loads\PrescribedDisplacement.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Loads\TemperatureLoad.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Properties\GeometricLine.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Properties\GeometricSurface.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Properties\LineMesh.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Properties\Material.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Properties\Support.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Create\Properties\SurfaceMesh.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Elements\Bars.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Elements\Edges.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Elements\Panels.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Loads\AreaTemperatureLoads.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Loads\AreaUniformlyDistributedLoads.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Loads\BarPointLoads.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Loads\BarTemperatureLoads.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Loads\BarUniformlyDistributedLoads.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Loads\BarVaryingDistributedLoads.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Loads\GravityLoads.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Loads\PointDisplacements.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Loads\PointForces.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Read\Results\ReadResults.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\CRUD\Update\UpdateObjects.cs

 - [x] Lusas_Toolkit\Lusas_Adapter\Private Helpers\GetAssignedObjects.cs

 - [x] Lusas_Toolkit\Lusas_Engine\Objects\Equality_Comparer\MeshComparer.cs

### LINKED ISSUE #295: BarUniformlyDistributedLoad not being pulled when set to local
#### Description:

Local distributed loads applied in Lusas do not get pulled.

#### Steps to reproduce:

1. Create local distributed load in Lusas 
2. Attempt to pull `BarDistributedLoad` 

#### Expected behaviour:

Allow user to pull `BarDistributedLoad` with `Local` settings.

### LINKED ISSUE #296: AreaUniformlyDistributedLoad not being pulled when set to local
#### Description:

Local distributed loads applied in Lusas do not get pulled.

#### Steps to reproduce:

1. Create local distributed load in Lusas 
2. Attempt to pull `AreaDistributedLoad` 

#### Expected behaviour:

Allow user to pull `AreaDistributedLoad` with `Local` settings.

### PR DESCRIPTION
**Pull request by peterjamesnugent**
### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/1033
https://github.com/BHoM/BHoM_Engine/pull/2106

 ### Issues addressed by this PR

Closes #293 
Closes #295 
Closes #296 

Aligning with changes in https://github.com/BHoM/BHoM/issues/1012

 

 ### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/ElfN7jbSoOBNoOxPwzCk8l8BvB3ttdUW1BffXhHWrrvp9A?e=PhMYb6

 ### Changelog

- Removes the use of `CustomData`, `MeshSettings1D`, `MeshSettings2D` and `LusasId` will now be stored as `Fragments`
- Migrates the `MeshSettings1D` and `MeshSettings2D` to the `BH.oM.Adapters.Lusas.Fragments` namespace
- Fixes bug where local `BarUniformlyDistributedLoad` and `AreaUniformlyDistributedLoad` were not being pulled from Lusas

### CODE REVIEW

File: Lusas_oM/Lusas_oM.csproj [hunk 1]
```diff
@@ -90,6 +90,9 @@
       <HintPath>C:\ProgramData\BHoM\Assemblies\BHoM.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Diffing_oM">
+      <HintPath>..\..\..\..\..\..\ProgramData\BHoM\Assemblies\Diffing_oM.dll</HintPath>
```
Comment (IsakNaslundBh) [inline Lusas_oM/Lusas_oM.csproj:94]:
```suggestion
      <HintPath>C:\ProgramData\BHoM\Assemblies\Diffing_oM.dll</HintPath>
```

**Review by IsakNaslundBh — CHANGES_REQUESTED** [review]
One reference needs fixing, see suggestion.

Think the rest looks good, but can not test as you know

**Comment by peterjamesnugent** [comment]
@StephennipBH I have added a versioning script, I've serailised some `MeshSettings1D` and `MeshSettings2D` as well as constructors on the canvas from today's Alpha. 

You will need to pull the latest from the branch and build the Versioning_Toolkit, then when you open the script the objects should be in the correct namespace. 

Let me know if you have any questions.

**Review by StephennipBH — CHANGES_REQUESTED** [review]
In pulling results, the same object is being pulled twice. (tested in v18 as suggested)

**Review by StephennipBH — APPROVED** [review]
The pulling object twice issue has been fixed. The rest of the PR work as intended. Changes approved.

**Pull request merged**